event_session/models/event_registration.py

```diff
@@ -53,7 +53,9 @@ def _check_seats_limit(self):
                 and session.seats_available < (1 if rec.state == "draft" else 0)
             ):
                 raise ValidationError(_("No more seats available for this session."))
-        return super(EventRegistration, regular_records)._check_seats_limit()
+        return super(
+            EventRegistration, regular_records
+        ).event_id._check_seats_availability()
```

Comment:
Why are u calling `super()` if you are calling a method of other model?

If this method does not exist on the model probably it's not needed.

Please check it.